pandas/core/arrays/datetimelike.py

```diff
@@ -52,6 +52,8 @@
 from pandas.tseries import frequencies
 from pandas.tseries.offsets import DateOffset, Tick
 
+DTScalarOrNaT = Union[DatetimeLikeScalar, NaTType]
```

Comment:
can't we add NaTType to the DatetimeLikeScalar?

Reply:
That would weaken the typing on the places where we currently use DatetimeLikeScalar